packages/core/src/State.ts

```diff
@@ -291,6 +291,8 @@ export function getPersistedState<TState extends AnyState>(
   const { configuration, transitions, tags, machine, children, ...jsonValues } =
     state;
 
+  const contextJSON = JSON.stringify(state.context);
```

Comment:
I don't think that we should be using `JSON.stringify` here. It's super lossy and doesn't handle some basic things - like `undefined`, `Map` etc.

When a user is using those kind of data structures they should be aware of the serializability issues and they need to implement their own serialize/deserialize logic (as in: we can't possibly handle that for them). 

The only thing that we should replace here are actor references as that's something the user can't do on their own.

Reply:
No longer being used